src/lib/pipeline.ml

```diff
@@ -316,16 +331,21 @@ let rec to_file_prefix:
       sprintf "%s-star-aligned" (to_file_prefix ?is sample)
     | Hisat (sample) ->
       sprintf "%s-hisat-aligned" (to_file_prefix ?is sample)
-    | Stringtie (sample) ->
-      sprintf "%s-stringtie" (to_file_prefix ?is sample)
+    | Stringtie (use_reference_gtf, sample) ->
+      sprintf "%s-%sstringtie"
+        (to_file_prefix ?is sample)
+        (match use_reference_gtf with
+        | `Yes -> "yesgtf"
```

Comment:
1) I think the only sane values here are `Yes` and `No`. The `If_available` option is just a recipe for mysterious failures downstream. 

2) ...so, since I only would want to use `Yes` and `No`, this becomes a custom boolean reimplemented with different names. Which, to me, is more confusing than just using a boolean directly. 

I'm voting for defaulting to reference guided and raising an exception early (before the job hits the Ketrew server) if the reference GTF isn't available (forcing the user to knowingly and manually select reference-free assembly).